### ISSUE TITLE
Empty unitPrice after add twice the same article to the cart

### ISSUE THREAD

**Issue by joenilson** [Sylius/Sylius#11491]
**Sylius version affected**: 1.7.3

**Description**  
When is added a listener to the sylius.order_item.post_add service it don't get some variables as unitPrice for an article that was added before to the cart.

**Steps to reproduce**  
- I add 10 units of 007M_black_elegance_jeans to the cart and it show me the getUnitPrice() with the correct value in the listener attached to the sylius.order_item.post_add.
- I return to the shop and go again to the 007M_black_elegance_jeans article and try to add again other 10 units, there is where i get the getUnitPrice() function in zero and other fields comes empty too.

**Comment by lchrusciel** [comment]
Hey Joe,

thanks for your report, however, I'm not exactly sure, how to reproduce the issue, you have described. May it be a problem with Doctrine proxy and data may be not fetched yet from the database? Are you able to provide a failing scenario for your case? Please consider that our official support channels are: 
 * [Slack](sylius.com/slack)
 * [StackOverflow](https://stackoverflow.com/search?tab=newest&q=sylius)
 * [Forum](https://forum.sylius.com/)

You can read more about them here: https://docs.sylius.com/en/latest/book/support/index.html

**Comment by vvasiloi** [comment]
Hey @lchrusciel, the issue was opened after we discussed it on Slack and I successfully reproduced it and identified it's root cause.
The bug is very simple to reproduce following the steps mentioned by @joenilson.
The cause of this bug is that the order item attached to the event  at https://github.com/Sylius/Sylius/blob/936e113123033fe2bc1ab62db1cc49c49af27407/src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php#L76 is always a new resource created at https://github.com/Sylius/Sylius/blob/936e113123033fe2bc1ab62db1cc49c49af27407/src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php#L44
This object is attached to the order, populated with data and persisted only when a product is first added to the cart.
After this order item object is created and the form is validated, then all other work is delegated to the order modifier at https://github.com/Sylius/Sylius/blob/936e113123033fe2bc1ab62db1cc49c49af27407/src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php#L76 which will only use this objects quantity to update existing order item:
https://github.com/Sylius/Sylius/blob/936e113123033fe2bc1ab62db1cc49c49af27407/src/Sylius/Component/Order/Modifier/OrderModifier.php#L52-L60
I see two options of solving this:
1. If an order item for current variant exists, then use it for "add to cart" command, otherwise create a new one.
2. Use the order object to find the persisted order item and use it for event.
